src/coupling/balance.cc

```diff
@@ -160,10 +160,27 @@ void Balance::allocate(unsigned int n_loc_dofs,
 	ASSERT(!allocation_done_, "Attempt to allocate Balance object multiple times.");
 	// Max. number of regions to which a single dof can contribute.
 	// TODO: estimate or compute this number directly (from mesh or dof handler).
-	const int n_bulk_regs_per_dof = min(10, (int)regions_.bulk_size());
+	const int n_bulk_regs_per_dof = min(10, (int)mesh_->region_db().bulk_size());
 	const unsigned int n_quant = quantities_.size();
-	const unsigned int n_bdr_reg = regions_.boundary_size();
-	const unsigned int n_blk_reg = regions_.bulk_size();
+	const unsigned int n_bdr_reg = mesh_->region_db().boundary_size();
+	const unsigned int n_blk_reg = mesh_->region_db().bulk_size();
+
+
+	// construct vector of regions of boundary edges
+    for (unsigned int loc_el = 0; loc_el < mesh_->get_el_ds()->lsize(); loc_el++)
```

Comment:
Could this loop be removed? It is missing in master and it is not moved neither.